### PR TITLE
ScrollView - extract top & bottom pocket components

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

**Review by novsstation — DISMISSED** [review]

File: js/renovation/ui/scroll_view/topPocket.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+export const viewFunction = (viewModel: TopPocket): JSX.Element => {
+  const {
+    topPocketRef,
+  } = viewModel;
+
+  return (
+    <div className={SCROLLVIEW_TOP_POCKET_CLASS} ref={topPocketRef}>
+      <div className={SCROLLVIEW_PULLDOWN}>
+        <div className={SCROLLVIEW_PULLDOWN_IMAGE_CLASS} />
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/topPocket.tsx]: https://github.com/DevExpress/DevExtreme/pull/15791/files?file-filters%5B%5D=.tsx#diff-d5a2474127579b204ca53ac74ae066581f6a9ad34e2b34fc2a7233e3719b5132R28
 
This code is copied from somewhere:

        <div className={SCROLLVIEW_PULLDOWN_IMAGE_CLASS} />
        <div className={SCROLLVIEW_PULLDOWN_INDICATOR_CLASS} />
        <div className={SCROLLVIEW_PULLDOWN_TEXT_CLASS}>
          <div className={SCROLLVIEW_PULLDOWN_VISIBLE_TEXT_CLASS}>Pull down to refresh...</div>
          <div>Release to refresh...</div>
          <div>Refreshing...</div>
        </div>
 
let's delete from toDelete renovated parts:
https://github.com/DevExpress/DevExtreme/blob/21_1/js/renovation/ui/scroll_view/_toDelete/ui.scroll_view.native.pull_down.js#L48

But be carefull,  _renderPullDown contains many non renovated code

File: js/renovation/ui/scroll_view/topPocket.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        <div className={SCROLLVIEW_PULLDOWN_INDICATOR_CLASS} />
+        <div className={SCROLLVIEW_PULLDOWN_TEXT_CLASS}>
+          <div className={SCROLLVIEW_PULLDOWN_VISIBLE_TEXT_CLASS}>Pull down to refresh...</div>
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/topPocket.tsx]: SCROLLVIEW_PULLDOWN_VISIBLE_TEXT_CLASS - this class is not added into markup:

```
        const $text = this._$pullDownText = $('<div>').addClass(SCROLLVIEW_PULLDOWN_TEXT_CLASS);

        this._$pullingDownText = $('<div>').text(this.option('pullingDownText')).appendTo($text);
        this._$pulledDownText = $('<div>').text(this.option('pulledDownText')).appendTo($text);
        this._$refreshingText = $('<div>').text(this.option('refreshingText')).appendTo($text);

```

It is managed by a code: https://github.com/DevExpress/DevExtreme/blob/21_1/js/renovation/ui/scroll_view/_toDelete/ui.scroll_view.native.pull_down.js#L76

Let's move it to a new card.

File: js/renovation/ui/scroll_view/topPocket.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+export class TopPocketProps {
+  @OneWay() someProp = false;
+}
+
+@Component({
+  view: viewFunction,
+})
+
+export class TopPocket extends JSXComponent(TopPocketProps) {
+  @Ref() topPocketRef!: RefObject<HTMLDivElement>;
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/topPocket.tsx]: Is it used? Let's add it later in a PR with the code that uses it

File: js/renovation/ui/scroll_view/topPocket.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+          <div>Refreshing...</div>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+@ComponentBindings()
+export class TopPocketProps {
+  @OneWay() someProp = false;
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/topPocket.tsx]: Typo?

File: js/renovation/ui/scroll_view/bottomPocket.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        <div className={SCROLLVIEW_REACHBOTTOM_INDICATOR_CLASS} />
+        <div className={SCROLLVIEW_REACHBOTTOM_TEXT_CLASS}>Updating...</div>
+      </div>
+    </div>
+  );
+};
+
+@ComponentBindings()
+export class BottomPocketProps {
+  @OneWay() someProp = false;
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/bottomPocket.tsx]: Typo?

File: js/renovation/ui/scroll_view/bottomPocket.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+@Component({
+  view: viewFunction,
+})
+
+export class BottomPocket extends JSXComponent(BottomPocketProps) {
+  @Ref() bottomPocketRef!: RefObject<HTMLDivElement>;
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/bottomPocket.tsx]: Is it used? Let's add it later in a PR with the code that uses it
I think we can delete the 
`const $bottomPocket = this._$bottomPocket = $('<div>').addClass(SCROLLVIEW_BOTTOM_POCKET_CLASS);`
line in this PR

File: js/renovation/ui/scroll_view/bottomPocket.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+export const viewFunction = (viewModel: BottomPocket): JSX.Element => {
+  const {
+    bottomPocketRef,
+  } = viewModel;
+
+  return (
+    <div className={SCROLLVIEW_BOTTOM_POCKET_CLASS} ref={bottomPocketRef}>
+      <div className={SCROLLVIEW_REACHBOTTOM_CLASS}>
+        <div className={SCROLLVIEW_REACHBOTTOM_INDICATOR_CLASS} />
+        <div className={SCROLLVIEW_REACHBOTTOM_TEXT_CLASS}>Updating...</div>
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/bottomPocket.tsx]: "Updating..." - let's renovate all texts later in a separate PR, simply don't delete the text assigning code from toDelete
And, I cannot find why this text should be 'Updating...'

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/renovation/ui/scroll_view/topPocket.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+  return (
+    <div className={SCROLLVIEW_TOP_POCKET_CLASS} ref={topPocketRef}>
+      <div className={SCROLLVIEW_PULLDOWN}>
+        <div className={SCROLLVIEW_PULLDOWN_IMAGE_CLASS} />
+        <div className={SCROLLVIEW_PULLDOWN_INDICATOR_CLASS} />
+        <div className={SCROLLVIEW_PULLDOWN_TEXT_CLASS}>
+          <div className={SCROLLVIEW_PULLDOWN_VISIBLE_TEXT_CLASS}>Pull down to refresh...</div>
+          <div>Release to refresh...</div>
+          <div>Refreshing...</div>
```
Comment (IgnatovDan) [inline js/renovation/ui/scroll_view/topPocket.tsx]: Pull down to refresh...
Release to refresh...
Refreshing...
Let's renovate all texts later in a separate PR, simply don't delete the text assigning code from toDelete

File: js/renovation/ui/scroll_view/topPocket.tsx (hunk 2 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view/topPocket.tsx]: We add this task into separate card - https://trello.com/c/SyrnCE0X/1605-scrollview-add-scrollviewpulldownvisibletextclass. We can't add element dynamically into the dom by code. I want to have the element which I will be control. Well, we do it later.

File: js/renovation/ui/scroll_view/bottomPocket.tsx (hunk 6 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view/bottomPocket.tsx]: Fix in 7d6265de52141a5f6ac3292fd8eca177347407e8

File: js/renovation/ui/scroll_view/topPocket.tsx (hunk 1 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view/topPocket.tsx]: Fix 3527be22a9377543d60df6a6071aec8f72421780

File: js/renovation/ui/scroll_view/bottomPocket.tsx (hunk 7 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view/bottomPocket.tsx]: Fix in 3527be22a9377543d60df6a6071aec8f72421780

File: js/renovation/ui/scroll_view/topPocket.tsx (hunk 3 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view/topPocket.tsx]: Fix in 35bf31969d74a3e0423e47b5cc9a22035698bce4

File: js/renovation/ui/scroll_view/topPocket.tsx (hunk 4 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view/topPocket.tsx]: Fix in 35bf31969d74a3e0423e47b5cc9a22035698bce4

File: js/renovation/ui/scroll_view/bottomPocket.tsx (hunk 5 above)
Comment (EugeniyKiyashko) [reply js/renovation/ui/scroll_view/bottomPocket.tsx]: Fix in 35bf31969d74a3e0423e47b5cc9a22035698bce4

**Review by novsstation — DISMISSED** [review]

**Review by KuznetsovVN — DISMISSED** [review]

**Review by novsstation — APPROVED** [review]

**Review by IgnatovDan — APPROVED** [review]

**Pull request merged**